client/src/pages/Events.tsx

```diff
@@ -1,20 +1,42 @@
 import SearchFilter from "../components/events-components/SearchFilter";
 import EventList from "../components/events-components/EventList";
 import { Button } from "@/components/ui/button";
+import { getAllEvents } from "@/utils/eventService";
+import { useEffect, useState } from "react";
+import type { EventType } from "@/utils/interfaces";
+import EventCreation from "@/components/events-components/EventCreation";
 
 const Events = () => {
+  const [events, setEvents] = useState<EventType[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [create, setCreate] = useState(false);
+
+  useEffect(() => {
+    const fetchEvents = async () => {
+      const retrievedEvents = await getAllEvents();
+      setEvents(retrievedEvents);
+      setLoading(false);
+    };
+    fetchEvents();
+  }, []);
+
+  if (loading) {
+    return <h1>Loading...</h1>;
+  }
```

Comment:
encourage to try `Suspense` for loading state:
https://react.dev/reference/react/Suspense